### PR TITLE
Reduce mean operator

### PR DESCRIPTION
**Pull request by scxiao**
should be merged after PR#292 (argmax/argmin)

### CODE REVIEW

File: src/include/migraphx/op/reduce_sum.hpp [hunk 1]
```diff
@@ -13,7 +14,7 @@ namespace op {
 
 struct reduce_sum
 {
-    std::vector<std::size_t> axes;
+    std::vector<int64_t> axes{};
```
Comment (pfultz2) [inline src/include/migraphx/op/reduce_sum.hpp]: The type should be `size_t` as the axes cannot be negative.

File: test/cpu_ops_test.cpp [hunk 2]
```diff
@@ -1763,4 +1763,79 @@ TEST_CASE(reduce_sum_test12)
     EXPECT(results_vector == gold);
 }
 
+TEST_CASE(reduce_mean_test1)
```
Comment (kahmed10) [inline test/cpu_ops_test.cpp]: can we rename these tests to reduce_mean_axis_{1,2,...}

Comment (kahmed10) [reply test/cpu_ops_test.cpp]: I notice sum tests above have the same naming convention, but feel free to change them too

Comment (kahmed10) [reply test/cpu_ops_test.cpp]: can we also get a test for no axis provided? The default should be to reduce all dimensions if no axis is provided, both for onnx and tf.

File: src/include/migraphx/op/reduce_sum.hpp (hunk 1 above)
Comment (scxiao) [reply src/include/migraphx/op/reduce_sum.hpp]: There are negative input in the bert model, so keep it as int64_t

File: test/cpu_ops_test.cpp (hunk 2 above)
Comment (scxiao) [reply test/cpu_ops_test.cpp]: Added and add shape test to cover code change

File: src/include/migraphx/op/reduce_sum.hpp (hunk 1 above)
Comment (kahmed10) [reply src/include/migraphx/op/reduce_sum.hpp]: Would it be better to reinterpret the negative input when parsing the protobuf? tf reduce_sum also takes in a negative value.

Comment (scxiao) [reply src/include/migraphx/op/reduce_sum.hpp]: I think this is just a design choice.

Comment (scxiao) [reply src/include/migraphx/op/reduce_sum.hpp]: Either way is fine, but we should be consistent across all our operators.

Comment (pfultz2) [reply src/include/migraphx/op/reduce_sum.hpp]: We usually use the negative indices in the operators.

File: test/op_shape_test.cpp [hunk 3]
```diff
@@ -390,37 +390,111 @@ TEST_CASE(softmax) { test_softmax_variations<migraphx::op::softmax>(); }
...
+        throws_shape(migraphx::op::argmin{4}, input);
+    }
+}
+
+template <class T>
+void test_reduce_ops()
+{
+    {
+        migraphx::shape input{migraphx::shape::float_type, {2, 3, 4, 5}};
+        expect_shape(migraphx::shape{migraphx::shape::float_type, {1, 1, 1, 1}}, T{}, input);
```
Comment (kahmed10) [inline test/op_shape_test.cpp]: This isn't the correct behavior in tensorflow. If no axis is provided, then the output should be a scalar. I believe it should be the same in onnx.

Comment (scxiao) [reply test/op_shape_test.cpp]: According to Paul's suggestion, our operator definition does not have the keep_dim attribute, and the default behavior is to keep dim. Then in parse onnx, we keep_dim is false, we call squeeze to convert it to a scalar.

Comment (kahmed10) [reply test/op_shape_test.cpp]: Ok, makes sense. If you want, you can leave a comment like: "operator keeps dims by default, unless user specifies in the protobuf"

Comment (scxiao) [reply test/op_shape_test.cpp]: Will add a comment for that.

**Pull request merged**